core/src/main/java/io/syndesis/core/Tokens.java

```diff
@@ -56,6 +68,65 @@ public static boolean isTokenExpired(String token) {
         }
     }
 
+    private static String getIssuer(String token) {
+        TokenVerifier verifier = TokenVerifier.create(token);
+        try {
+            return verifier.getToken().getIssuer();
+        } catch (VerificationException e) {
+            throw SyndesisServerException.launderThrowable(e);
+        }
+    }
+
+    public static String getProviderToken(String providerId) {
+        String keycloakTokenAsString = getAuthenticationToken();
+
+        String issuer = getIssuer(keycloakTokenAsString);
```

Comment:
The issuer should always be the same as the external URL to the keycloak server.